src/System.Windows.Forms.Primitives/src/Interop/Shell32/Interop.DragQueryFileW.cs

```diff
@@ -0,0 +1,47 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Runtime.InteropServices;
+using System.Text;
+
+internal static partial class Interop
+{
+    internal static partial class Shell32
+    {
+        [DllImport(Libraries.Shell32, ExactSpelling = true, CharSet = CharSet.Unicode)]
+        public static extern uint DragQueryFileW(IntPtr hDrop, uint iFile, StringBuilder lpszFile, uint cch);
+
+        public static uint DragQueryFileLongPath(IntPtr hDrop, uint iFile, StringBuilder lpszFile)
+        {
+            if (lpszFile == null || lpszFile.Capacity == 0 || iFile == 0xFFFFFFFF)
+            {
+                return DragQueryFileW(hDrop, iFile, lpszFile, (uint)lpszFile.Capacity);
```

Comment:
This will result in a NRE if `lpszFile == null`. The original code has obviously been buggy.
```suggestion
                return DragQueryFileW(hDrop, iFile, null, 0);
```